File: python/observer/WeatherRama/observer/displays/CurrentConditionDisplay.py
from observer.Observer import Observer
from observer.Display import Display
from subject.WeatherStation import WeatherStation

class CurrentConditionDisplay(Observer, Display):

    def __init__(self, weather_station):
        self.__temperature = 0
        self.__humidity = 0
        self.__pressure = 0
        self.__weather_station = WeatherStation()
        self.__weather_station = weather_station
        self.__weather_station.register_observer(self)

    def update(self, temperature, humidity, pressure):
        self.__temperature = temperature
        self.__humidity = humidity
        self.__pressure = pressure
        self.display()

    def display(self):
        print("\nCurrent Conditions:\n"
              + str(self.__temperature) + "F degree\n"
              + str(self.__humidity) + "% humidity\n"
              + str(self.__pressure) + " pressure\n")

File: C++/Observer/WeatherRama/observer/displays/CurrentConditionDisplay.cpp
#include "../Display.h"
#include "../Observer.h"
#include "../../subject/WeatherStation.cpp"
#include <iostream>

using namespace std;

class CurrentConditionDisplay : public Observer, public Display {

public:

    explicit CurrentConditionDisplay(WeatherStation* weatherStation) {
        this->weatherStation = weatherStation;
        this->weatherStation->registerObserver(this);
    }

    void update(float temperature, float humidity, float pressure) override {
        Temperature = temperature;
        Humidity = humidity;
        Pressure = pressure;
        display();
    }

    void display() override {
        cout << "\nCurrent Condition:" << endl;
        cout << Temperature << "F degree\n";
        cout << Humidity << "% humidity\n";
        cout << Pressure << " pressure\n";
    }

private:
    float Temperature{};
    float Humidity{};
    float Pressure{};

    WeatherStation* weatherStation = new WeatherStation();
};

